format/FlightSql.proto

```diff
@@ -1549,6 +1658,17 @@ message DoPutUpdateResult {
   int64 record_count = 1;
 }
 
+/*
+ * Explicitly cancel a running query.
+ *
+ * For distributed queries (ones that use GetFlightInfo->DoGet), this lets a
+ * single client explicitly cancel work across all clients, given server support.
```

Comment:
are there semantics that need to be considered relative to transcations?

Reply:
I'll update it to specify that the transaction is not rolled back/this is only to terminate reading of the result set (CC @jduo is that in line with what you were thinking?)

Reply:
This is what I was thinking in terms of read, but also think it should terminate a running write (same as in ODBC/JDBC).

Reply:
Thanks - updated. The semantics around cancelling a write are a little unclear, I suppose in autocommit mode it gets committed, and otherwise the client can commit or rollback.